src/Access/DiskAccessStorage.cpp

```diff
@@ -224,63 +225,57 @@ bool DiskAccessStorage::isPathEqual(const String & directory_path_) const
 }
 
 
-void DiskAccessStorage::clear()
-{
-    entries_by_id.clear();
-    for (auto type : collections::range(AccessEntityType::MAX))
-        entries_by_name_and_type[static_cast<size_t>(type)].clear();
-}
-
-
 bool DiskAccessStorage::readLists()
 {
-    clear();
+    std::vector<std::tuple<UUID, String, AccessEntityType>> ids_names_types;
 
-    bool ok = true;
     for (auto type : collections::range(AccessEntityType::MAX))
     {
-        auto & entries_by_name = entries_by_name_and_type[static_cast<size_t>(type)];
         auto file_path = getListFilePath(directory_path, type);
         if (!std::filesystem::exists(file_path))
         {
             LOG_WARNING(getLogger(), "File {} doesn't exist", file_path);
-            ok = false;
-            break;
+            return false;
         }
 
         try
         {
             for (const auto & [id, name] : readListFile(file_path))
-            {
-                auto & entry = entries_by_id[id];
-                entry.id = id;
-                entry.type = type;
-                entry.name = name;
-                entries_by_name[entry.name] = &entry;
-            }
+                ids_names_types.emplace_back(id, name, type);
         }
         catch (...)
         {
             tryLogCurrentException(getLogger(), "Could not read " + file_path);
-            ok = false;
-            break;
+            return false;
         }
     }
 
-    if (!ok)
-        clear();
-    return ok;
+    entries_by_id.clear();
+    for (auto type : collections::range(AccessEntityType::MAX))
+        entries_by_name_and_type[static_cast<size_t>(type)].clear();
+
+    for (const auto & [id, name, type] : ids_names_types)
```

Comment:
```suggestion
    for (auto & [id, name, type] : ids_names_types)
```
Now you can move `name`

Reply:
done